### PR TITLE
Update fraction.rb

### PR DESCRIPTION
**Pull request by atomski**

### CODE REVIEW

**Comment by atomski** [comment]
@nonkor, ready for review

File: fraction.rb [hunk 1]
```diff
@@ -1,17 +1,64 @@
 class Fraction
   def initialize(numerator, denominator)
-    if !(numerator.is_a?(Integer) && denominator.is_a?(Integer))
-      # TODO: create a proper exception for that issue
-      # TODO: handle case when one or both parameters are not Integer but could be easily transformed into it
-      # by using #to_i
-      raise 'At least some arguments are incorrect: ' + numerator.inspect + ', ' + denominator.inspect
+    if ! numerator.is_a?(Integer) && denominator.is_a?(Integer)
```
Comment (nonkor) [inline fraction.rb]: You should manage these validations separately: one check for `numerator` and other for `denominator`.
And makes sense to create new kind of exception for it.

File: fraction.rb [hunk 2]
```diff
@@ -1,17 +1,64 @@
...
+       @denominator = @denominator /b
+       @numerator = @numerator /b
+     end
+      b=b-1
+=end
+    Fraction.new(@numerator / @numerator.gcd(@denominator), @denominator / @denominator.gcd(@numerator))
+  end
+
+  def view
+      puts "#{@numerator}/#{@denominator}"
```
Comment (nonkor) [inline fraction.rb]: Do not use puts here. The purpose of that method to prepare it for showing. You should not call `puts` right now, only in a place where you will need it. Like:

``` ruby
f = Fraction.new(1, 3)
puts f.view
```

File: fraction.rb [hunk 3]
```diff
@@ -1,17 +1,64 @@
...
+  end
+
+  def +(a)
```
Comment (nonkor) [inline fraction.rb]: What is `a`? Please, change it to more descriptive argument.

File: fraction.rb [hunk 4]
```diff
@@ -1,17 +1,64 @@
...
+    Fraction.new(@numerator*a.denominator+a.numerator*@denominator, @denominator*a.denominator).simplify
+  end
+
+  def *(a)
+    Fraction.new(@numerator = @numerator*a.numerator, @denominator = @denominator*a.denominator).simplify
```
Comment (nonkor) [inline fraction.rb]: Do not create such long lines, it's hard to read. It's just bad style.
Better to do something like:

``` ruby
num, den = @numerator * a.numerator, @denominator * a.denominator
Fraction.new(num, den).simplify
```

And be aware about mutating existing data.
Try to call smth like:

``` ruby
f1 = Fraction.new(2, 3)
f2 = Fraction.new(5, 7)
f3 = f1 * f2
f1.view  # => 10/21
```

Any thoughts why `f1` was changed too?

File: fraction.rb [hunk 5]
```diff
@@ -1,17 +1,64 @@
...
+  end
+
+  def /(a)
+    Fraction.new(@numerator*a.denominator, @denominator*a.numerator).simplify
+  end
+
+  def -(a)
+    simplify
```
Comment (nonkor) [inline fraction.rb]: This and next line do **nothing**.

File: fraction.rb [hunk 6]
```diff
@@ -1,17 +1,64 @@
...
+    @numerator
+  end
+
+  def denominator
+    @denominator
+  end
+
+  def simplify
+
+=begin
```
Comment (nonkor) [inline fraction.rb]: Why did you leave this?

**Comment by nonkor** [comment]
@atomski 
So, please refactor due to a first part of my comments. After fixing this I'll return to review.
And one more meaningful thing: you have to describe changes which you do in commit properly. Please, [read here](http://chris.beams.io/posts/git-commit/) about good practice of writing git commit message.

**Comment by atomski** [comment]
@nonkor please see the changes

File: fraction.rb [hunk 7]
```diff
@@ -1,17 +1,61 @@
+
```
Comment (nonkor) [inline fraction.rb:1]: Delete this. Do you check diff before committing?

File: fraction.rb [hunk 8]
```diff
@@ -1,17 +1,61 @@
...
 class Fraction
-  def initialize(numerator, denominator)
-    if !(numerator.is_a?(Integer) && denominator.is_a?(Integer))
-      # TODO: create a proper exception for that issue
-      # TODO: handle case when one or both parameters are not Integer but could be easily transformed into it
-      # by using #to_i
-      raise 'At least some arguments are incorrect: ' + numerator.inspect + ', ' + denominator.inspect
+  def check_fraction(numerator, denominator)
+    if ! numerator.is_a?(Integer) && (denominator.is_a?(Integer) && denominator>0)
```
Comment (nonkor) [inline fraction.rb:4]: It's still one cumulative validation for two numbers. [I've asked](https://github.com/nonkor/Betlab-ruby-courses/pull/1#discussion_r70066317) about splitting them and creation custom Exception classes.

File: fraction.rb [hunk 9]
```diff
@@ -1,17 +1,61 @@
...
+  def denominator
+    @denominator
+  end
+
+  def simplify
+    Fraction.new(@numerator / @numerator.gcd(@denominator), @denominator / @denominator.gcd(@numerator))
+  end
+
+  def view
+      "#{@numerator}/#{@denominator}"
```
Comment (nonkor) [inline fraction.rb:28]: wrong indentation

File: fraction.rb [hunk 10]
```diff
@@ -1,17 +1,61 @@
...
+  def numerator
+    @numerator
+  end
+
+  def denominator
+    @denominator
+  end
+
+  def simplify
+    Fraction.new(@numerator / @numerator.gcd(@denominator), @denominator / @denominator.gcd(@numerator))
```
Comment (nonkor) [inline fraction.rb:24]: why not:

``` ruby
gdc = @numerator.gcd(@denominator)
Fraction.new(@numerator / gdc, @denominator / gdc)
```

?
Less lines and easier to read.

File: fraction.rb [hunk 11]
```diff
@@ -1,17 +1,61 @@
...
+  end
+
+  def view
+      "#{@numerator}/#{@denominator}"
+  end
+
+  def +(fract)
+    num = @numerator*fract.denominator+fract.numerator*@denominator
```
Comment (nonkor) [inline fraction.rb:32]: spaces, spaces, spaces!

File: fraction.rb [hunk 12]
```diff
@@ -1,17 +1,61 @@
...
+    den = @denominator*fract.denominator
+    Fraction.new(num, den).simplify
+  end
+
+  def *(fract)
+    num = @numerator*fract.numerator
```
Comment (nonkor) [inline fraction.rb:38]: spaces, spaces, spaces!

File: fraction.rb [hunk 13]
```diff
@@ -1,17 +1,61 @@
...
+    den = @denominator*fract.denominator
+    Fraction.new(num, den).simplify
+  end
+
+  def /(fract)
+    num = @numerator*fract.denominator
```
Comment (nonkor) [inline fraction.rb:44]: spaces, spaces, spaces!

File: fraction.rb [hunk 14]
```diff
@@ -1,17 +1,61 @@
...
+    den = @denominator*fract.numerator
+    Fraction.new(num, den).simplify
+  end
+
+  def -(fract)
+    num = @numerator*fract.denominator-fract.numerator*@denominator
```
Comment (nonkor) [inline fraction.rb:50]: spaces, spaces, spaces!

File: fraction.rb [hunk 15]
```diff
@@ -1,17 +1,61 @@
...
+    Fraction.new(num, den).simplify
+  end
 end
 
-# puts Fraction.new('s_numerator', 's_denominator') #=> fail
-# puts Fraction.new('s_numerator', 1) #=> fail
-# puts Fraction.new(2, '1') #=> fail
-puts Fraction.new(1, 2) #=> ok
+
+fraction1 = Fraction.new(3,12)
```
Comment (nonkor) [inline fraction.rb:57]: please, add one case with simplifying (+ with one negative number in numerator or denominator)

**Comment by nonkor** [comment]
@atomski great work!

Please, check new comments and apply them. I'd rather see one good commit instead of two with the same message. Please, [care about description](https://github.com/nonkor/Betlab-ruby-courses/pull/1#issuecomment-231350837).

**Pull request closed without merging**